lib/config/presets/index.ts

```diff
@@ -154,11 +154,43 @@ export function parsePreset(input: string): ParsedPreset {
   return { presetSource, presetPath, packageName, presetName, params };
 }
 
+export const removedPresets = {
+  ':automergeBranchMergeCommit': ':automergeBranch',
+  ':automergeBranchPush': ':automergeBranch',
+  ':base': 'config:base',
+  ':app': 'config:js-app',
+  ':js-app': 'config:js-app',
+  ':library': 'config:js-lib',
+  ':masterIssue': ':dependencyDashboard',
+  ':masterIssueApproval': ':dependencyDashboardApproval',
+  ':unpublishSafe': 'npm:unpublishSafe',
+  'config:application': 'config:js-app',
+  'config:base-js': 'config:base',
+  'config:library': 'config:js-lib',
+  'default:automergeBranchMergeCommit': ':automergeBranch',
+  'default:automergeBranchPush': ':automergeBranch',
+  'default:base': 'config:base',
+  'default:app': 'config:js-app',
+  'default:js-app': 'config:js-app',
+  'default:library': 'config:js-lib',
+  'default:unpublishSafe': 'npm:unpublishSafe',
+  'helpers:oddIsUnstable': null,
+  'helpers:oddIsUnstablePackages': null,
+};
+
 export async function getPreset(
   preset: string,
   baseConfig?: RenovateConfig
 ): Promise<RenovateConfig> {
   logger.trace(`getPreset(${preset})`);
+  // Check if the preset has been removed or replaced
+  const newPreset = removedPresets[preset];
+  if (newPreset) {
+    return getPreset(newPreset, baseConfig);
+  }
+  if (newPreset === null) {
+    return {};
```

Comment:
I think we should warn if an old preset is used.

Reply:
⬆️ ❓

Reply:
I didn't do that as we will migrate such unused presets out in future with a PR anyway. I don't think we need to warn people if e.g. we have a preset to enable an experimental feature and later it becomes default.